quesma/queryparser/aggregation_parser.go

```diff
@@ -366,12 +366,20 @@ func (cw *ClickhouseQueryTranslator) parseFieldFromScriptField(queryMap QueryMap
 		logger.WarnWithCtx(cw.Ctx).Msgf("source is not a string, but %T, value: %v", sourceRaw, sourceRaw)
 	}
 
-	// source must look like "doc['field_name'].value.getHour()" or "doc['field_name'].value.hourOfDay"
+	// a) source must look like "doc['field_name'].value.getHour()" or "doc['field_name'].value.hourOfDay"
 	wantedRegex := regexp.MustCompile(`^doc\['(\w+)']\.value\.(?:getHour\(\)|hourOfDay)$`)
 	matches := wantedRegex.FindStringSubmatch(source)
 	if len(matches) == 2 {
 		return model.NewFunction("toHour", model.NewColumnRef(matches[1])), true
 	}
+
+	// b) source: "if (doc['field_name_1'].value == doc['field_name_2'].value") { return 1; } else { return 0; }"
+	wantedRegex = regexp.MustCompile(`^if \(doc\['(.*)\.value']\.value == doc\['(.*)\.value'].value\) \{ \n  return 1; \n} else \{ \n  return 0; \n}$`)
```

Comment:
We eventually need to unify this with rest of painless parsing, but good for now.